lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -33,20 +33,25 @@ void circt::firrtl::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
   auto dstType = dstFType.dyn_cast<FIRRTLBaseType>();
   auto srcType = srcFType.dyn_cast<FIRRTLBaseType>();
 
-  // Special Connects
-  if (!dstType) {
-    if (dstFType.isa<RefType>() &&
-        !dstFType.cast<RefType>().getType().hasUninferredWidth())
-      builder.create<RefConnectOp>(dst, src);
-    else // Other types, give up and leave a connect
-      builder.create<ConnectOp>(dst, src);
+  // References use ref.assign.  Types should match, leave to verifier if not.
+  if (isa<RefType>(dstFType)) {
+    builder.create<RefDefineOp>(dst, src);
     return;
   }
 
   // If the types are the exact same we can just connect them.
-  // Strict connect does not allow uninferred widths.
-  if (dstType == srcType && !dstType.hasUninferredWidth()) {
-    builder.create<StrictConnectOp>(dst, src);
+  if (dstFType == srcFType) {
+    // Strict connect does not allow uninferred widths.
```

Comment:
strictconnect will soon allow all identical types, inferred or not.  But it doesn't yet.

Reply:
Reverted to the new behavior, but also good to know.